File: lang/go/src/muse/algorithms/number_theory/Coprimality/Coprimality.go
package Coprimality

import "muse/algorithms/number_theory/GreatestCommonDivisor"

func ReduceToBinaryGCD(m int64, n int64) bool {
    return GreatestCommonDivisor.IterativeBinaryGCD(m, n) == 1
}

func ReduceToEuclidean(m int64, n int64) bool {
    return GreatestCommonDivisor.IterativeEuclidean(m, n) == 1
}

File: archive/nodejs/muse/algorithms/number_theory/Coprimality.js
import * as GreatestCommonDivisor from './GreatestCommonDivisor.js';

export const reduceToBinaryGCD = (m, n) => {
  return GreatestCommonDivisor.iterativeBinaryGCD(m, n) === 1;
};

export const reduceToEuclidean = (m, n) => {
  return GreatestCommonDivisor.iterativeEuclidean(m, n) === 1;
};
